CI: Integration Test - TestRestoredPort

## CI failure

Link: https://github.com/cilium/cilium/actions/runs/12867119980/job/35871110616

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/proxy/proxyports   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestRestoredPort (0.01s)

time="2025-01-20T11:47:21.003558766Z" level=info msg="Adding new proxy port rules for cilium-http-egress:17266" id=cilium-http-egress subsys=proxy
    proxyports_test.go:282: 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/proxy/proxyports/proxyports_test.go:282
        	Error:      	Should be zero, but was 17266
        	Test:       	TestRestoredPort
```

--- a/pkg/proxy/proxyports/proxyports_test.go
+++ b/pkg/proxy/proxyports/proxyports_test.go
@@ -8,6 +8,7 @@ import (
 	"os"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/cilium/cilium/pkg/envoy"
@@ -278,8 +279,8 @@ func TestRestoredPort(t *testing.T) {
 	require.Zero(t, pp.nRedirects)
 
 	// wait for port reuse wait to pass
-	time.Sleep(time.Millisecond)
-	require.Zero(t, pp.ProxyPort)
+	// waiting time is set up to 1s (instead of exactly 1ms) to avoid potential flake in CI
+	require.Eventually(t, func() bool { return assert.Zero(t, pp.ProxyPort) }, time.Second, time.Millisecond)
 	require.False(t, pp.configured)
 	require.False(t, pp.acknowledged)
 

